src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,182 @@
+                () -> registrationService.register(user1));
+    }
+
+    @Test
+    void register_smallLogin_notOk() {
+        user1.setLogin("");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+        user1.setLogin(" ");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+        user1.setLogin("q");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+        user1.setLogin("qwe");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+        user1.setLogin("qwert");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+    }
+
+    @Test
+    void register_normalLogin_Ok() throws InvalidDataException {
+        Assertions.assertEquals(user1, registrationService.register(user1));
+        user2.setLogin("login10");
+        Assertions.assertEquals(user2, registrationService.register(user2));
+        user3.setLogin("userloginuserloginuserlogin");
+        Assertions.assertEquals(user3, registrationService.register(user3));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user1.setPassword(null);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+    }
+
+    @Test
+    void register_smallPassword_notOk() {
+        user1.setPassword("");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+        user1.setPassword(" ");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+        user1.setPassword("1");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+        user1.setPassword("123");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+        user1.setPassword("12345");
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1));
+    }
+
+    @Test
+    void register_normalPassword_Ok() throws InvalidDataException {
+        Assertions.assertEquals(user1, registrationService.register(user1));
+        user2.setPassword("1234567");
+        Assertions.assertEquals(user2, registrationService.register(user2));
+        user3.setPassword("1234567891011121314151617");
+        Assertions.assertEquals(user3, registrationService.register(user3));
+    }
+
+    @Test
+    void register_addingExistingUser_notOk() throws InvalidDataException {
+        registrationService.register(user1);
+        Assertions.assertNull(registrationService.register(user1));
+    }
```

Comment:
you can't rely on register to test register
![image](https://github.com/mate-academy/jv-registration-validation/assets/49767049/f73226f4-75db-4460-ab41-e2230d9f20b5)